tsdb/engine/tsm1/engine.go

```diff
@@ -150,6 +154,7 @@ func (e *Engine) Close() error {
 	// Lock now and close everything else down.
 	e.mu.Lock()
 	defer e.mu.Unlock()
+	e.done = nil // Ensures that the channel will not be closed again.
```

Comment:
Setting the channel to `nil` causes problems with code using default case in the selector such as:

https://github.com/influxdata/influxdb/blob/master/tsdb/engine/tsm1/engine.go#L454-L458

When it's `nil`, if falls through to the `default` clause.